Mono.Addins/Mono.Addins.Database/AddinFileSystemExtension.cs

```diff
@@ -201,22 +201,18 @@ public virtual void DeleteFile (string filePath)
 			File.Delete (filePath);
 		}
 
-		internal void CleanupReflector()
+		protected internal virtual void CleanupReflector()
 		{
-			var disposable = reflector as IDisposable;
-			if (disposable != null)
-				disposable.Dispose ();
+			using (reflector as IDisposable) { }
 		}
-		
+
 		/// <summary>
 		/// Gets a value indicating whether this <see cref="Mono.Addins.Database.AddinFileSystemExtension"/> needs to be isolated from the main execution process
 		/// </summary>
 		/// <value>
 		/// <c>true</c> if requires isolation; otherwise, <c>false</c>.
 		/// </value>
-		public virtual bool RequiresIsolation {
-			get { return true; }
-		}
+		public virtual bool RequiresIsolation { get; set; } = true;
```

Comment:
Isn't this an API break? People overriding need to override the set now too.

Reply:
Yup! Curious how many consumers this will break. Not sure how to access the risk vs. reward here.

Reply:
It’s the opposite - Mikayla inherits from this class just to override the setter, and with this change she won’t have to.